crux-mir-comp/src/Mir/Compositional.hs

```diff
@@ -49,23 +48,23 @@ compositionalOverrides _symOnline cs name cfg
         return $ MethodSpecBuilder msb
 
   | hasInstPrefix ["crucible", "method_spec", "raw", "builder_add_arg"] explodedName
-  , Empty :> MethodSpecBuilderRepr :> MirReferenceRepr _tpr <- cfgArgTypes cfg
+  , Empty :> MethodSpecBuilderRepr :> MirReferenceRepr <- cfgArgTypes cfg
   , MethodSpecBuilderRepr <- cfgReturnType cfg
   = Just $ bindFnHandle (cfgHandle cfg) $ UseOverride $
     mkOverride' "method_spec_builder_add_arg" MethodSpecBuilderRepr $ do
         RegMap (Empty :> RegEntry _tpr (MethodSpecBuilder msb)
-            :> RegEntry (MirReferenceRepr tpr) argRef) <- getOverrideArgs
-        msb' <- msbAddArg tpr argRef msb
+            :> RegEntry MirReferenceRepr argRef) <- getOverrideArgs
+        msb' <- msbAddArg {-tpr-} (error "TODO RGS") argRef msb
```

Comment:
Haven't looked at this too closely, but my guess is you'll need to get the `Mir.Ty` and convert it with `tyToRepr`.  It looks like you do have access to the `Collection` here, so you can look up the `DefId` `name` to get an `Instance`, then inspect the `Substs` of that `Instance` to get the necessary `Ty`.  I think the code in crucible-mir responsible for applying the various `TransCustom` overrides has to do essentially the same thing, so it might be helpful to track that down as a reference.

Reply:
I've implemented your suggestion, which is sufficient to compute the `TypeRepr`. With this, I've confirmed locally that the `crux-mir-comp` test suite passes with these changes.